migrator/README.md

```diff
@@ -132,7 +133,7 @@ it manipulates and for converting it.
 #### Create or upgrade the schema of a table.
 
 ```go
-pkgSchema.ApplySchemaForTable(context.Background(), databases.PostgresDB, <schema>)
+pgutils.CreateTableFromModel(context.Background(), gormDB, frozenSchema.CreateTableCollectionsStmt)
```

Comment:
If this is the call we are going to recommend we should make it clear that for the frozen version of the schema we need a copy of the entire schema and not just the gorm struct.  Also I think we should call out that for data manipulation migrations we should not need to freeze the schema nor make this call.

Reply:
There should indeed be a section covering the case of data manipulation migrations. I think these will be common migrations.
I agree there is no need to call `pgutils.CreateTableForModel` in data manipulation migrations, but I think these should rely on a frozen schema and on frozen store code, though.

Reply:
regarding the frozen schema for data migrations, sure, but we should we make a new copy?.  For instance, if we make a data migration in 4.0, should we make a copy of the schema or just use the most recent frozen one?  Such as using the v73 frozen schema for a 4.0 data only migration instead of copying the same v73 schema into the v4.0 frozen schemas.

Reply:
Indeed, the last known frozen state of the schema / store should be sufficient for data manipulation migrations.
I'm also not sure release numbering is the right approach for schema versions. I could well imagine the same table changing multiple times during the same release to accommodate for the needs of different features or projects.
IMHO, the schema version should be incremented each time a datamodel change is submitted. Impacted stores could then be pinned to the frozen changed schema (that could avoid duplication like the `permissionset` store present in `m_168_to_m_169` and in `m_169_to_m_170`).

Reply:
That is a good point.  I agree, aside from the initial v73 frozen set, future schema changes should be frozen in the specific migration package itself.  That will make it easier to find and follow what is happening.

Reply:
That is a good point.  I agree, aside from the initial v73 frozen set, future schema changes should be frozen in the specific migration package itself.  That will make it easier to find and follow what is happening.

Reply:
I moved the related notes to an separate session and add more details. Thanks for the feedback.